code/modules/hydroponics/hydroponics_mutations.dm

```diff
@@ -0,0 +1,529 @@
+			3;	"plusstat_light_tolerance", \
+			3;	"plusstat_nutrient&water_consumption", \
+			S.yield != -1 && !S.harvest_repeat ? 0.4 : 0;	"toggle_repeatharvest",\
+			)
+		if("SomethingWeird")
+			mutation_type = pick(\
+			S.biolum ? 10 : 0;"biolum_changecolor",\
+			S.biolum ? 1 : 10;"trait_biolum",\
+			S.juicy ? 0.5 : 5;"trait_juicy", \
+			S.juicy == 1 ? 10 : 2 ;"trait_slippery", \
+			S.thorny ? 0.2 : 5;"trait_thorns",\
+			S.parasite ? 0.2 : 5;"trait_parasitic",\
+			S.carnivorous ? 0.1 : 5;"trait_carnivorous",\
+			S.carnivorous == 1 ? 8 : 2;"trait_carnivorous2",\
+			S.ligneous ? 0.2 : 5;"trait_ligneous"
+			)
+		if("SomethingBizarre")
+			mutation_type = pick(\
+			4;"chemical_exotic", \
+			6;"fruit_exotic", \
+			2;"change_appearance", \
+			S.teleporting ? 0.2 : 5;"trait_teleporting", \
+			S.spread ? 0.1 : 2;"trait_creepspread",
+			)
+		if("SomethingBad")
+			mutation_type = pick(\
+			3;"tox_increase", \
+			2;"weed_increase", \
+			2;"pest_increase", \
+			5;"stunt_growth", \
+			)
+		if("SomethingAwful")
+			mutation_type = pick(\
+			S.hematophage ? 0.2 : 5;"trait_hematophage", \
+			5;"randomize_light", \
+			5;"randomize_temperature", \
+			2;"breathe_aliengas", \
+			S.yield != -1 && S.harvest_repeat ? 2 : 0;	"toggle_repeatharvest",\
+			)
+		if("SomethingDangerous")
+			mutation_type = pick(\
+			4;"spontaneous_creeper", \
+			1;"spontaneous_kudzu", \
+			S.spread == 1 ? 5 : 1;"trait_vinespread",
+			S.stinging ? 0.2 : 4;"trait_stinging", \
+			1;"exude_dangerousgas", \
+			S.alter_temp ? 0.2 : 2;"change_roomtemp"
+			)
+	return mutation_type
+
+/obj/machinery/portable_atmospherics/hydroponics/proc/show_mutation_message()
+	visible_message("<span class='notice'>\The [seed.display_name] quivers!</span>")
+
+/obj/machinery/portable_atmospherics/hydroponics/proc/check_for_divergence(var/modified = 0)
+	// We need to make sure we're not modifying one of the global seed datums.
+	// If it's not in the global list, then no products of the line have been
+	// harvested yet and it's safe to assume it's restricted to this tray.
+	if(!isnull(plant_controller.seeds[seed.name]))
+		seed = seed.diverge(modified)
+
+/obj/machinery/portable_atmospherics/hydroponics/proc/get_ratio(var/severity, var/list/softcaps, var/list/hardcaps, var/input)
+	var/i = min(Ceiling(severity/5), 5)
+	var/lerp_factor = (severity % 5) /5
+	var/softcap = Lerp(softcaps[i], softcaps[i+1], lerp_factor)
+	var/hardcap = Lerp(hardcaps[i], hardcaps[i+1], lerp_factor)
+	. = linear_extrapolation(input, softcap, hardcap)
+	world << "Caps: [softcap]-[hardcap] / Final Ratio: [.]"
+	return
+
+/obj/machinery/portable_atmospherics/hydroponics/proc/apply_mut(var/mutation_type, var/severity)
+
+	// Check if we should even bother working on the current seed datum.
+	if(seed.immutable > 0) return
+
+	check_for_divergence()
+
+	world << "So here we are about to apply this mutation [mutation_type]"
+	switch(mutation_type)
+		if("code_explanation")
```

Comment:
Probably yes, though I'm not smart enough to do it and have never created something similar